# Week 1 - Data Cleaning & Exploration

Name: Muhammad Muneeb Sohail

Internship:Data Science Internship

Task:Applicant Dataset Cleaning using Pandas

In [1]:
import pandas as pd
import numpy as np

In [4]:
from faker import Faker
import random

fake = Faker()

In [5]:
domains = [
    "Web Development",
    "AI",
    "Graphic Design",
    "Data Science",
    "Cyber Security"
]

status = [
    "Selected",
    "Rejected",
    "Under Review"
]

rows=[]

for i in range(100):
    rows.append({
        "Applicant Name":fake.name(),
        "Email":fake.email(),
        "Phone":fake.phone_number(),
        "Domain Applied":random.choice(domains),
        "University":fake.company(),
        "Application Date":fake.date_between("-120d","today"),
        "Status":random.choice(status)
    })

df=pd.DataFrame(rows)

In [6]:
df.loc[5,"Email"]=None
df.loc[8,"University"]=None
df.loc[10,"Status"]=None
df.loc[15,"Applicant Name"]=None

df.loc[20,"Domain Applied"]="web dev"
df.loc[21,"Domain Applied"]="WEB DEV"
df.loc[22,"Domain Applied"]="web development"

df.loc[30,"Application Date"]="12/05/2026"
df.loc[31,"Application Date"]="2026/06/01"

df=df._append(df.iloc[4],ignore_index=True)
df=df._append(df.iloc[9],ignore_index=True)

In [7]:
df.to_csv("applicants.csv",index=False)

In [8]:
df=pd.read_csv("applicants.csv")

In [9]:
df.shape
df.head()
df.info()
df.describe(include="all")
df.isnull().sum()
df.duplicated().sum()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102 entries, 0 to 101
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Applicant Name    101 non-null    object
 1   Email             101 non-null    object
 2   Phone             102 non-null    object
 3   Domain Applied    102 non-null    object
 4   University        101 non-null    object
 5   Application Date  102 non-null    object
 6   Status            101 non-null    object
dtypes: object(7)
memory usage: 5.7+ KB


np.int64(2)

handling missing values now

In [10]:
df=df.dropna(subset=["Applicant Name"])

In [11]:
df["Email"]=df["Email"].fillna("not_provided@example.com")

In [12]:
df["University"]=df["University"].fillna("Not Specified")

In [13]:
df["Status"]=df["Status"].fillna("Under Review")

applicant name is critical so i remove it university was not so i make it not specified

In [14]:
df.duplicated().sum()

np.int64(2)

In [15]:
df=df.drop_duplicates()

In [16]:
df=df.drop_duplicates(subset=["Email"])

for spaces remove

In [17]:
df["Applicant Name"]=df["Applicant Name"].str.strip()

df["Domain Applied"]=df["Domain Applied"].str.strip()

df["Status"]=df["Status"].str.strip()

for lower case

In [18]:
df["Domain Applied"]=df["Domain Applied"].str.lower()

In [19]:
mapping={
"web dev":"Web Development",
"web development":"Web Development",
"WEB DEV":"Web Development",
"ai":"AI",
"graphic design":"Graphic Design",
"data science":"Data Science",
"cyber security":"Cyber Security"
}

In [20]:
df["Domain Applied"]=df["Domain Applied"].replace(mapping)

In [21]:
df["Domain Applied"].unique()

array(['Web Development', 'Graphic Design', 'Data Science',
       'Cyber Security', 'AI'], dtype=object)

In [22]:
df["Status"]=df["Status"].str.title()

In [23]:
df["Application Date"]=pd.to_datetime(
    df["Application Date"],
    errors="coerce"
)

In [24]:
df["Phone"]=(
    df["Phone"]
    .astype(str)
    .str.replace("-","",regex=False)
    .str.replace(" ","",regex=False)
)

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 99 entries, 0 to 99
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Applicant Name    99 non-null     object        
 1   Email             99 non-null     object        
 2   Phone             99 non-null     object        
 3   Domain Applied    99 non-null     object        
 4   University        99 non-null     object        
 5   Application Date  97 non-null     datetime64[ns]
 6   Status            99 non-null     object        
dtypes: datetime64[ns](1), object(6)
memory usage: 6.2+ KB


# Data Quality Summary

Initial Dataset

102 rows
7 columns

Problems Found

- Missing values
- Duplicate rows
- Inconsistent Domain names
- Wrong date datatype
- Phone formatting issues

Cleaning Performed

- Removed rows with missing Applicant Name
- Filled missing Email
- Filled University with Not Specified
- Filled Status with Under Review
- Removed duplicate records
- Standardized Domain names
- Converted dates into datetime
- Standardized phone numbers

Final Dataset

Clean
No duplicates
Consistent formatting
Correct datatypes

In [26]:
df.to_csv("applicants_cleaned.csv",index=False)